File: back/Dto/ProductSearchInputDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace back.Dto
{
    public class ProductSearchInputDto
    {
        public string? ProductName { get; set; }
        public int? QuantityFrom { get; set; }
        public int? QuantityTo { get; set; }
        public string? UnitOfMeasure { get; set; }
        public string? Status { get; set; }
        public DateTime? DateFrom { get; set; }
        public DateTime? DateTo { get; set; }
        public long? CategoryId { get; set; }
    }
}

File: front/src/app/_Dto/ProductSearchInputDto.ts
export class ProductSearchInputDto {
    productName: string;
    quantityFrom: number;
    quantityTo: number;
    unitOfMeasure: string;
    status: string;
    dateFrom: string;
    dateTo: string;
    categoryId: number;
    constructor() {
            this.productName = "";
            this.quantityFrom = 0;
            this.quantityTo = 0;
            this.unitOfMeasure = "";
            this.status = "";
            this.dateFrom = "";
            this.dateTo = "";
            this.categoryId = 0;
    }
}